### ISSUE TITLE
`method-hidden` false positive with `functools.lru_cache` in `__init__`

### ISSUE THREAD

**Issue by nyuszika7h** [PyCQA/pylint#7405]
### Bug description

When using `@lru_cache` on an instance method, flake8-bugbear throws a B019 warning, therefore I do this in `__init__` instead:
```py
self.get_proxy = lru_cache(get_proxy)
```

However, this causes pylint to raise a false positive warning on `def get_proxy(self, region)`.

### Command used

```shell
pylint foo/services/__init__.py
```

### Pylint output

```shell
method-hidden: An attribute defined in foo.services line 60 hides this method
```

### Expected behavior

There should be no warning.

### Pylint version

```shell
pylint 2.15.0     
astroid 2.12.5
Python 3.10.4 (main, Apr  4 2022, 14:54:08) [GCC 10.2.1 20210110]
```

### OS / Environment

Debian 10

**Comment by Pierre-Sassoulas** [comment]
I don't think this is a false positive as the method really is hidden, disabling the pylint message will probably make reader of your code to be more aware of this fact.
